lib/compat/wordpress-6.1/block-editor-settings.php

```diff
@@ -0,0 +1,164 @@
+ */
+function gutenberg_get_block_editor_settings( $settings ) {
+	// Set what is the context for this data request.
+	$context = 'other';
+	if (
+		defined( 'REST_REQUEST' ) &&
+		REST_REQUEST &&
+		isset( $_GET['context'] ) &&
+		'mobile' === $_GET['context']
+	) {
+		$context = 'mobile';
+	}
+
+	if ( 'other' === $context ) {
+		global $wp_version;
+		$is_wp_5_8 = version_compare( $wp_version, '5.8', '>=' ) && version_compare( $wp_version, '5.9', '<' );
+		$is_wp_5_9 = version_compare( $wp_version, '5.9', '>=' ) && version_compare( $wp_version, '6.0-beta1', '<' );
+		$is_wp_6_0 = version_compare( $wp_version, '6.0-beta1', '>=' );
+
+		// Make sure the styles array exists.
+		// In some contexts, like the navigation editor, it doesn't.
+		if ( ! isset( $settings['styles'] ) ) {
+			$settings['styles'] = array();
+		}
+
+		// Remove existing global styles provided by core.
+		$styles_without_existing_global_styles = array();
+		foreach ( $settings['styles'] as $style ) {
+			if (
+				( $is_wp_5_8 && ! gutenberg_is_global_styles_in_5_8( $style ) ) || // Can be removed when plugin minimum version is 5.9.
+				( $is_wp_5_9 && ! gutenberg_is_global_styles_in_5_9( $style ) ) || // Can be removed when plugin minimum version is 6.0.
+				( $is_wp_6_0 && ( ! isset( $style['isGlobalStyles'] ) || ! $style['isGlobalStyles'] ) )
+			) {
+				$styles_without_existing_global_styles[] = $style;
+			}
+		}
+
+		// Recreate global styles.
+		$new_global_styles = array();
+		$presets           = array(
+			array(
+				'css'            => 'variables',
+				'__unstableType' => 'presets',
+				'isGlobalStyles' => true,
+			),
+			array(
+				'css'            => 'presets',
+				'__unstableType' => 'presets',
+				'isGlobalStyles' => true,
+			),
+		);
+		foreach ( $presets as $preset_style ) {
+			$actual_css = gutenberg_get_global_stylesheet( array( $preset_style['css'] ) );
+			if ( '' !== $actual_css ) {
+				$preset_style['css'] = $actual_css;
+				$new_global_styles[] = $preset_style;
+			}
+		}
+
+		if ( WP_Theme_JSON_Resolver::theme_has_support() ) {
+			$block_classes = array(
+				'css'            => 'styles',
+				'__unstableType' => 'theme',
+				'isGlobalStyles' => true,
+			);
+			$actual_css    = gutenberg_get_global_stylesheet( array( $block_classes['css'] ) );
+			if ( '' !== $actual_css ) {
+				$block_classes['css'] = $actual_css;
+				$new_global_styles[]  = $block_classes;
+			}
+		} elseif( current_theme_supports( 'wp-block-styles' ) ) {
+			// If there is no `theme.json` file, but the theme opts in to block styles, ensure base layout styles are available.
+			$block_classes = array(
+				'css'            => 'base-layout-styles',
+				'__unstableType' => 'theme',
+				'isGlobalStyles' => true,
+			);
+			$actual_css    = gutenberg_get_global_stylesheet( array( $block_classes['css'] ) );
+			if ( '' !== $actual_css ) {
+				$block_classes['css'] = $actual_css;
+				$new_global_styles[]  = $block_classes;
+			}
```

Comment:
This is the only new section in this file — it is to ensure that we output base layout styles for classic themes, if they opt-in to the core block styles.